_search-plugins/search-pipelines/rerank-processor.md

```diff
@@ -0,0 +1,116 @@
+
+### The `ml_opensearch` reranker type
+
+The `ml_opensearch` reranker type is designed to work with the cross-encoder model provided by OpenSearch. For this reranker type, specify the following fields.
+
+Field  | Data type | Description
+:--- | :---  | :--- 
+`ml_opensearch` | Object | Provides the rerank processor with model information. Required.
+`ml_opensearch.model_id` | String | The model ID for the cross-encoder model. Required. For more information, see [Using ML models]({{site.url}}{{site.baseurl}}/ml-commons-plugin/using-ml-models/).
+`context.document_fields` | Array | An array of document fields that specifies the fields from which to retrieve context for the cross-encoder model. Required.
+
+## Example 
+
+The following example demonstrates using a search pipeline with a `rerank` processor.
+
+### Creating a search pipeline
+
+The following request creates a search pipeline with a `rerank` response processor:
+
+```json
+PUT /_search/pipeline/rerank_pipeline
+{
+  "response_processors": [
+    {
+      "rerank": {
+        "ml_opensearch": {
+          "model_id": "gnDIbI0BfUsSoeNT_jAw"
+        },
+        "context": {
+          "document_fields": [ "title", "text_representation"]
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Using a search pipeline
+
+Combine an OpenSearch query with an `ext` object that contains the query context for the LLM. Provide the `query_text` that will be used to rerank the results:
+
+```json
+POST /_search?search_pipeline=rerank_pipeline
+{
+  "query": {
+    "match": {
+      "text_representation": "Where is Albuquerque?"
+    }
+  },
+  "ext": {
+    "rerank": {
+      "query_context": {
+        "query_text": "Where is Albuquerque?"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Instead of specifying `query_text`, you can provide a full path to the field that contains text to use for reranking. For example, if you specify a subfield `query` in the `text_representation` object, specify its path in the `query_text_path` parameter:
+
+```json
+POST /_search?search_pipeline=rerank_pipeline
+{
+  "query": {
+    "match": {
+      "text_representation": {
+        "query": "Where is Albuquerque?"
+      }
+    }
+  },
+  "ext": {
+    "rerank": {
+      "query_context": {
+        "query_text_path": "query.match.text_representation.query"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The `query_context` object contains the following fields. 
+
+Field name  | Description
+:--- | :---  
+`query_text` | The natural language text of the question using which you want to rerank the search results. Exactly one of `query_text` and `query_text_path` (not both) is required.
```

Comment:
Can the last sentence be something like "Either `query_text` or `query_text_path` is required. These fields may not be used simultaneously"?